### PR TITLE
feat: Set import source when creating My Collection artworks + refactoring

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-953]

### Description

This PR
- sets import source when creating My Collection artworks to `MY_Collection`
- renames `myCollectionAddArtworkMutation` to `myCollectionCreateArtworkMutation` 
- renames `myCollectionEditArtworkMutation` to `myCollectionUpdateArtworkMutation` 
to align with conventions and the name of the mutations: `myCollectionCreateArtwork` & `myCollectionUpdateArtwork`.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Set import source when creating My Collection artworks - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-953]: https://artsyproduct.atlassian.net/browse/CX-953?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tests.tsx [hunk 1]
```diff
@@ -139,6 +139,49 @@ describe("MyCollectionArtworkForm", () => {
...
+          }
+        `)
+
+        mockEnvironment.mock.resolveMostRecentOperation(() => ({
+          data: {},
+        }))
+        await flushPromiseQueue()
+
+        const createArtworkOperation = mockEnvironment.mock.getMostRecentOperation()
+        expect(createArtworkOperation.request.variables).toMatchInlineSnapshot(`
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tests.tsx:161]: Finally adding the missing peace to this test.

File: src/app/Scenes/MyCollection/mutations/myCollectionCreateArtwork.ts [hunk 2]
```diff
@@ -32,7 +35,7 @@ export function myCollectionAddArtwork(input: myCollectionAddArtworkMutationVari
         }
       `,
       variables: {
-        input,
+        input: { importSource: "MY_COLLECTION", ...input },
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/mutations/myCollectionCreateArtwork.ts:38]: This is the only real change of this PR.

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tests.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tests.tsx:161]: **praise:** good choice with going with an inline snapshot 👏

**Review by MounirDhahri — APPROVED** [review]
🎉

**Comment by MounirDhahri** [comment]
@olerichter00 you're on fire today 😄

**Pull request merged**